### PR TITLE
feat(rpc): Add `z_gettreestate` gRPC tests

### LINKED ISSUE #4351: Add GetTreeState gRPC test
In https://github.com/ZcashFoundation/zebra/pull/4253 we added tests for grpc wallet methods, however we left a few TODOs in the main file (zebrad/tests/common/lightwalletd/wallet_grpc_test.rs) for missing tests. 

Add a test for GetTreeState

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation

Zebra should have tests for all RPCs. This PR adds some tests for the `z_gettreestate` RPC.

## Solution

The tests check all the fields in the RPC response against manually defined static values.

Closes #4351.

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

### CODE REVIEW

File: zebrad/tests/common/lightwalletd/wallet_grpc_test.rs [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-//! This test requires a cached chain state that is partially synchronized, i.e., it should be a
-//! few blocks below the network chain tip height. It also requires a lightwalletd data dir in sync
-//! with the cached chain state.
+//! This test requires a cached chain state that is partially synchronized,
+//! i.e., it should be a few blocks below the network chain tip height. It also
+//! requires a lightwalletd data dir in sync with the cached chain state.
 //!
-//! Current coverage of all available rpc methods according to `CompactTxStreamer`:
+//! Current coverage of all available rpc methods according to
+//! `CompactTxStreamer`:
```
Comment (teor2345) [inline zebrad/tests/common/lightwalletd/wallet_grpc_test.rs:9]: Nitpick: For future PRs, can you configure your editor so it doesn't reflow paragraphs?

It's ok to change edited lines, but changing unrelated lines makes it hard to see which docs have actually changed.

(Feel free to just close this comment after you've read it!)

**Review by teor2345 — APPROVED** [review]
Looks good, thanks!

File: zebrad/tests/common/lightwalletd/wallet_grpc_test.rs (hunk 1 above)
Comment (upbqdn) [reply zebrad/tests/common/lightwalletd/wallet_grpc_test.rs:9]: Sure, thanks for the suggestion.

**Comment by upbqdn** [comment]
Can I do anything about https://github.com/ZcashFoundation/zebra/runs/6546283264?check_suite_focus=true#step:7:11392?

**Comment by conradoplg** [comment]
> Can I do anything about https://github.com/ZcashFoundation/zebra/runs/6546283264?check_suite_focus=true#step:7:11392?

This is https://github.com/ZcashFoundation/zebra/issues/4456
We might want to update the checkpoints. There is not issue for that yet, but feel free to take a look into that

**Review by teor2345 — APPROVED** [review]
We're still waiting on #4415, so I think we should merge this now, and fix any bugs we find later.

**Comment by teor2345** [comment]
@mergifyio refresh

**Pull request merged**